File: src/com/colacelli/irclib/connection/listeners/OnChannelNoticeMessageListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.connection.Connection;
import com.colacelli.irclib.messages.ChannelNoticeMessage;

public interface OnChannelNoticeMessageListener extends Listener {
    Type TYPE = Type.CHANNEL_NOTICE;

    void onChannelNoticeMessage(Connection connection, ChannelNoticeMessage message);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/main/kotlin/com/colacelli/irclib/connection/listeners/OnChannelNoticeMessageListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.connection.Connection
import com.colacelli.irclib.messages.ChannelNoticeMessage

interface OnChannelNoticeMessageListener : Listener {
    fun onChannelNoticeMessage(connection: Connection, message: ChannelNoticeMessage)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.CHANNEL_NOTICE
    }
}
